src/AccessLogEntry.cc

```diff
@@ -96,16 +96,19 @@ AccessLogEntry::syncNotes(HttpRequest *req)
         assert(notes == req->notes());
 }
 
-const char *
+Ident::User
 AccessLogEntry::getClientIdent() const
 {
-    if (tcpClient)
-        return tcpClient->rfc931;
-
-    if (cache.rfc931 && *cache.rfc931)
-        return cache.rfc931;
+    return acceptedClientConnection ? acceptedClientConnection->ident : std::nullopt;
+}
 
-    return nullptr;
+void
+AccessLogEntry::setClientIdent(const char *name)
+{
+    if (acceptedClientConnection)
+        acceptedClientConnection->setIdent(name);
+    else
+        debugs(30, DBG_IMPORTANT, "ERROR: cannot set ident: client connection is missing");
```

Comment:
This is a rather odd error: in a bug-free code, a connection (that we are using for IDENT query) should not go missing (when we receive the response). Can we refactor so that this disappearance becomes impossible in a bug-free code and then Assure() acceptedClientConnection existence?

It also feels a bit odd to task AccessLogEntry to set information that belongs to a Connection, especially when ALE might eventually store many Connections -- we do not want to create a wrapper method for each of those Connections. Perhaps there is a way for the IDENT lookup code to store and adjust the right Connection object instead (without knowing that it is storing and adjusting ACLE::acceptedClientConnection)?

Reply:
>  there is a way for the IDENT lookup code to store and adjust the right Connection object instead

I think it is possible, done (e701d15).